cmake/multi_image.cmake

```diff
@@ -56,6 +56,21 @@ function(zephyr_add_external_image_from_source name sourcedir)
   set(${NAME}_CMAKE_BINARY_DIR ${CMAKE_BINARY_DIR}/${name})
   set(${NAME}_PROJECT_BINARY_DIR ${CMAKE_BINARY_DIR}/${name}/zephyr)
 
+  get_cmake_property(VARIABLES_CACHED CACHE_VARIABLES)
+  get_cmake_property(VARIABLES              VARIABLES)
+
+  unset(image_cmake_args)
+  foreach(var_name
+      ${VARIABLES}
+      ${VARIABLES_CACHED}
+      )
+    if("${var_name}" MATCHES "^${name}_(.+)") # TODO: How expensive?
+      list(APPEND image_cmake_args
+        -D${CMAKE_MATCH_1}=${${var_name}}
```

Comment:
I like the approach, and it surely open for a lot of interesting configuration options, but we need to be a little careful.

This allows a user to pass: `cmake -DBOARD=nrf9160_pca10090ns -DSPM_BOARD=nrf52840_pca10056 ..` which is definitely not something we want (currently).
The `SPM_BOARD` will become `BOARD` for sub-image and thus there will be two `-DBOARD` arguments on CMake execution for SPM, in this example.

Also it opens up for passing in another `CMAKE_MAKE_PROGRAM` than what is used for app build process.
All this can lead to a lot of strange behavior or combinations that we cannot guarentee will work.
This is of course the users responsibility, but it might be more user friendly to specify what can be passed in, and what is documented / tested, instead of basically allowing everything.

Should this this be combined with some white-list / black-list of variables allowed ?

Reply:
I considered a whitelist, but decided not to do so, as I consider the cost of maintaining this list (with support for options not listed in the in-tree list), of options and documenting them to be more expensive than the odd accidental aliasing issue.

But if time shows that users often accidentally alias options or if the regex proves to be expensive we can reconsider.

Reply:
Fine with me.